### ISSUE TITLE
How to use heapprofd on android application memory usage.

### ISSUE THREAD

**Issue by NitzDKoder** [catapult-project/perfetto#2]
Hi Folks,

https://github.com/catapult-project/perfetto/blob/master/tools/heap_profile
https://www.perfetto.dev/#/heapprofd?id=viewing-the-data

Looking to run this on android Q device and do memory profile. Documents are minimal. 

@segfaulthunter Please share details on how to run on android Application and what windows environment setup required for the same.?

 **ex:**

tools/heap_profile --name system_server

system_server is package name of the application?

Thanks

**Comment by fmayer** [comment]
Thanks for your interest.

As far as I understand you are trying to run the tools/heap_profile script on Windows. Is that correct?

If that is the case, we sadly do not support Windows at this point in time.

**Comment by NitzDKoder** [comment]
@NitzDKoder yes run  tools/heap_profile on windows machine and profile android application.

So what other machine " tools/heap_profile " is supported? linux and mac machine?

Thanks

**Comment by fmayer** [comment]
Yes, tools/heap_profile is currently supported on Linux and MacOS.

**Comment by fmayer** [comment]
Hey!

You can now record Heap Profiles from the Perfetto UI directly. Head over to https://ui.perfetto.dev/#!/record?p=memory, press "Add Device +" to pair to your device connected via USB.  Then enter the name of the target process in "Names or pids of the processes to track". Make sure you are allowed to profile the process (as per https://docs.perfetto.dev/#/heapprofd?id=target-processes). Also set the "Max duration" to the duration you want to profile for in "Recording settings".

Then a diamond/triangle will appear. Press that to view the profile. You can also download the profile there in a pprof compatible format, which you can view using speedscope.app or the pprof CLI (https://github.com/google/pprof).
